blobs/howto/python/python-v12/crud_datalake.py

```diff
@@ -24,287 +24,180 @@
+    # </Snippet_RenameDirectory>
+
+    # -------------------------------------------------
+    # Delete directory method
+    # -------------------------------------------------
+
+    # <Snippet_DeleteDirectory>
+    def delete_directory(self, directory_client: DataLakeDirectoryClient):
         directory_client.delete_directory()
-    except Exception as e:
-     print(e) 
-# </Snippet_DeleteDirectory>
-
-# ------------------------------------------------
-# Upload a file to a directory
-# ------------------------------------------------
-
-# <Snippet_UploadFile>
-def upload_file_to_directory():
-    try:
+    # </Snippet_DeleteDirectory>
 
-        file_system_client = service_client.get_file_system_client(file_system="my-file-system")
+    # ------------------------------------------------
+    # Upload a file to a directory
+    # ------------------------------------------------
 
-        directory_client = file_system_client.get_directory_client("my-directory")
+    # <Snippet_UploadFile>
+    def upload_file_to_directory(self, directory_client: DataLakeDirectoryClient, local_path: str, file_name: str):
+        file_client = directory_client.create_file(file_name)
         
-        file_client = directory_client.create_file("uploaded-file.txt")
-        local_file = open("C:\\file-to-upload.txt",'r')
+        with open(file=os.path.join(local_path, file_name), mode="rb") as data:
+            file_contents = data.read()
 
-        file_contents = local_file.read()
-
-        file_client.append_data(data=file_contents, offset=0, length=len(file_contents))
+        file_client.append_data(
+            data=file_contents, offset=0, length=len(file_contents))
 
         file_client.flush_data(len(file_contents))
+    # </Snippet_UploadFile>
 
-    except Exception as e:
-      print(e) 
-# </Snippet_UploadFile>
-
-# ------------------------------------------------
-# Upload a file to a directory in bulk
-# ------------------------------------------------
+    # ------------------------------------------------
+    # Upload a file to a directory in bulk
+    # ------------------------------------------------
 
-# <Snippet_UploadFileBulk>
-def upload_file_to_directory_bulk():
-    try:
+    # <Snippet_UploadFileBulk>
+    def upload_file_to_directory_bulk(self, directory_client: DataLakeDirectoryClient, local_path: str, file_name: str):
+        file_client = directory_client.get_file_client(file_name)
 
-        file_system_client = service_client.get_file_system_client(file_system="my-file-system")
-
-        directory_client = file_system_client.get_directory_client("my-directory")
-        
-        file_client = directory_client.get_file_client("uploaded-file.txt")
+        with open(file=os.path.join(local_path, file_name), mode="rb") as data:
+            file_contents = data.read()
 
-        local_file = open("C:\\file-to-upload.txt",'r')
+        file_client.upload_data(file_contents, overwrite=True)
```

Comment:
The `upload_data` API takes in a stream as well so it's probably better to recommend just passing the file stream directly to append rather than reading all the contents into memory first. Internally `upload_data` works like `upload_blob` where it will split the upload in 100 MiB chunks (by default) and only load 100 MiB into memory at a time.

```py
with open(file=os.path.join(local_path, file_name), mode="rb") as data:
     file_client.upload_data(data, overwrite=True)
```